core/src/main/scala/ai/forestflow/utils/SourceStorageProtocols.scala

```diff
@@ -85,7 +93,44 @@ object SourceStorageProtocols extends StrictLogging {
   }
 
   case object S3 extends EnumVal {
-    def downloadDirectoryImpl(remotePath: String, localDirectory: File, fqrv: FQRV, sslVerify: Boolean): Unit = ???
+
+    private implicit val actorSystem: ActorSystem = ActorSystem("Application-s3-stream")
```

Comment:
There is already an actor system. This is trying to create another one, on the fly, every time an S3 download is called. The actor system is generally created once for the entire application life-cycle so creating it here doesn't make sense from that context not to mention that one already exists.